### PR TITLE
Update sample app 4.18.0

### PR DESCRIPTION
**Pull request by Slorq**

### CODE REVIEW

File: Carthage Sample/Carthage-sample-app.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -434,7 +450,7 @@
 				DEVELOPMENT_TEAM = 68F5DK9B95;
 				FRAMEWORK_SEARCH_PATHS = "$(inherited)";
 				INFOPLIST_FILE = "Carthage-sample-app/Info.plist";
-				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.4;
```
Comment (wangjiejacques) [inline Carthage Sample/Carthage-sample-app.xcodeproj/project.pbxproj:453]: Why change this?

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
+
+import ContentsquareErrorAnalysisModule
+import UIKit
```
Comment (wangjiejacques) [inline Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift:3]: @Slorq Did you test if all the error analysis samples work as expected?

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+/// This class showcases how to use Error Analysis to trace HTTP network errors
+class ErrorAnalysisViewController: UIViewController {
```
Comment (wangjiejacques) [inline Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift:6]: I know it's done like this previously, but I was wondering if we can use the same file for all samples.
Let me check with the team if it's a good idea.

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        request.httpMethod = method
+        return request
+    }
+
+    private func printResponse(data: Data?) {
+        guard let data = data, let dataString = String(data: data, encoding: .utf8) else {
+            return
+        }
+        NSLog("ErrorAnalysis: Response, \(dataString)")
+    }
```
Comment (wangjiejacques) [inline Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift:62]: We may also need to update the README file, for example, add "error analysis" into these 2 lines.
> Here you can find iOS sample applications that will show you how to use Contentsquare SDK and different ways to integrate it.

> If you wish to learn more about the Contentsquare SDK you can find the documentation [here](https://docs.contentsquare.com/ios/)

And for each sample, there is a README.md inside, we also need to update it.

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift (hunk 3 above)
Comment (wangjiejacques) [reply Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift:6]: Let's keep it as it is for now.

File: Carthage Sample/Carthage-sample-app.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Slorq) [reply Carthage Sample/Carthage-sample-app.xcodeproj/project.pbxproj]: I added it to use the async methods, but I reverted it and added some conditionals based on the available iOS

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift (hunk 3 above)
Comment (Slorq) [reply Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift]: Removed duplicates

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift (hunk 4 above)
Comment (Slorq) [reply Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift]: Updated

File: Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift (hunk 2 above)
Comment (Slorq) [reply Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift]: You mean all the methods from `URLSession`?

Comment (wangjiejacques) [reply Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift]: All the [IBAction]s in [ErrorAnalysisViewController.swift](https://github.com/ContentSquare/iOS-sample-app/pull/36/files/a5f408df38e4cbae20d86d010d0e6f3ca8184bcf#diff-9c37896219f25ae455d3c25de278aec65c885c24371ecd4add3833cb7942d0b7).

Comment (Slorq) [reply Static Framework Sample/static-framework-sample-app/Showcases/ErrorAnalysis/ErrorAnalysisViewController.swift]: Yes, I did test all the IBActions in all the sample apps

**Review by wangjiejacques — APPROVED** [review]

**Pull request merged**